src/Control/SearchEditor.php

```diff
@@ -167,25 +167,45 @@ protected function applyChanges(Filter\Rule $rule, array &$values, array $path =
             }
 
             $newValue = $this->popKey($values, $identifier . '-value');
+            $oldValue = $rule->getValue();
             if ($newValue !== null && $rule->getValue() !== $newValue) {
                 $rule->setValue($newValue);
             }
 
             $newOperator = $this->popKey($values, $identifier . '-operator');
             if ($newOperator !== null && QueryString::getRuleSymbol($rule) !== $newOperator) {
+                $value = $rule->getValue();
+                $column = $rule->getColumn();
                 switch ($newOperator) {
                     case '=':
-                        return Filter::equal($rule->getColumn(), $rule->getValue());
+                        if (is_string($value) && strpos($value, '*') !== false) {
+                            return Filter::similar($column, $value);
+                        }
+
+                        return Filter::equal($column, $value);
                     case '!=':
-                        return Filter::unequal($rule->getColumn(), $rule->getValue());
+                        if (is_string($value) && strpos($value, '*') !== false) {
+                            return Filter::unlike($column, $value);
+                        }
+
+                        return Filter::unequal($column, $value);
                     case '>':
-                        return Filter::greaterThan($rule->getColumn(), $rule->getValue());
+                        return Filter::greaterThan($column, $value);
                     case '>=':
-                        return Filter::greaterThanOrEqual($rule->getColumn(), $rule->getValue());
+                        return Filter::greaterThanOrEqual($column, $value);
                     case '<':
-                        return Filter::lessThan($rule->getColumn(), $rule->getValue());
+                        return Filter::lessThan($column, $value);
                     case '<=':
-                        return Filter::lessThanOrEqual($rule->getColumn(), $rule->getValue());
+                        return Filter::lessThanOrEqual($column, $value);
+                }
+            }
+
+            $value = $rule->getValue();
+            if (is_string($value) && strpos($value, '*') !== false && $oldValue !== $value) {
```

Comment:
Check `$oldValue !== $value` first, the other checks don't run then if that's not true.